src/uu/numfmt/src/format.rs

```diff
@@ -392,12 +392,20 @@ fn format_and_print_whitespace(s: &str, options: &NumfmtOptions) -> Result<()> {
 
             print!("{}", format_string(field, options, implicit_padding)?);
         } else {
+            // the -z option converts an initial \n into a space
+            let prefix = if options.zero_terminated && prefix.starts_with('\n') {
+                print!(" ");
+                &prefix[1..]
+            } else {
+                prefix
+            };
```

Comment:
This is probably just some implementation detail, but one of the GNU compatibility tests depends on this behavior. As per the guidelines, I haven't examined the GNU implementation for `numfmt`, but here are some manual tests I did for compatibility:
```
$ printf "A\nB 1001 C\x00D E\n2002 F\x00" | ./src/numfmt -z --field=3 --to=si | cat --show-all # official test
A B 1.1k C^@D E 2.1k F^@⏎
$ printf "A \nB 1001 C\x00D E\n2002 F\x00" | ./src/numfmt -z --field=3 --to=si | cat --show-all
A $
B 1.1k C^@D E 2.1k F^@⏎
$ printf "A\n B 1001 C\x00D E\n2002 F\x00" | ./src/numfmt -z --field=3 --to=si | cat --show-all
A  B 1.1k C^@D E 2.1k F^@⏎
```
The above output is from the GNU version of `numfmt`, but it's identical to the uutils version output (with this patch of course).